assets/js/components/Num/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Formatted number component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { numFmt } from '../../util';
+
+export default function Num( {
```

Comment:
I think this component as well as the related `Decimal` component are a bit overkill and too low-level for components. It makes sense to create reusable components to create a specific UI, but these two components are purely for formatting a single value, that could easily be achieved by just calling `numFmt`.

Rather than introducing yet another layer around formatting numbers, let's use the function directly.

Reply:
To me the component/JSX form is easier to read and a bit more natural semantically:

![image](https://user-images.githubusercontent.com/1621608/110774935-05792580-8267-11eb-8888-c577898f8fda.png)

I'll update it to use `numFmt` though, not a problem.